pkg/adapterManager/managerDispatchBenchmark_test.go

```diff
@@ -0,0 +1,244 @@
+
+How to run the test:
+# make sure the code builds and works with go tools and not just with bazel.
+
+Command:
+go test -run XXXX -bench .
+
+Output: (2017-04-27)
+BenchmarkOneSimpleAspect-12     	   10000	    113880 ns/op
+Benchmark50SimpleAspect-12      	    1000	   1410848 ns/op
+BenchmarkOneComplexAspect-12    	   10000	    157771 ns/op
+Benchmark50ComplexAspect-12     	     500	   2757074 ns/op
+PASS
+ok  	istio.io/mixer/pkg/adapterManager	6.202s
+*/
+
+const (
+	minimalGlobalCnfg = `
+subject: namespace:ns
+revision: "2022"
+adapters:
+  - name: noopMetricKindAdapter
+    kind: metrics
+    impl: noopMetricKindAdapter
+
+manifests:
+  - name: istio-proxy
+    revision: "1"
+    attributes:
+    - name: source.name
+      value_type: STRING
+    - name: target.name
+      value_type: STRING
+    - name: response.code
+      value_type: INT64
+    - name: api.name
+      value_type: STRING
+    - name: api.method
+      value_type: STRING
+
+metrics:
+  - name: request_count
+    kind: COUNTER
+    value: INT64
+    description: request count by source, target, service, and code
+    labels:
+      source: 1 # STRING
+      target: 1 # STRING
+      service: 1 # STRING
+      method: 1 # STRING
+      response_code: 2 # INT64
+  - name: request_latency
+    kind: COUNTER
+    value: DURATION
+    description: request latency by source, target, and service
+    labels:
+      source: 1 # STRING
+      target: 1 # STRING
+      service: 1 # STRING
+      method: 1 # STRING
+      response_code: 2 # INT64
+`
+	srvcCnfgConstInitialSection = `
+subject: namespace:ns
+rules:
+- selector: true
+  aspects:
+`
+	srvcCnfgSimpleAspect = `
+  - kind: metrics
+    adapter: noopMetricKindAdapter
+    params:
+      metrics:
+      - descriptorName: request_count
+        value: response.code | 100
+        labels:
+          source: source.name | "one"
+          target: target.name | "one"
+          service: api.name | "one"
+          method: api.method | "one"
+          response_code: response.code | 111
+`
+
+	srvcCnfgComplexAspect = `
+  - kind: metrics
+    adapter: noopMetricKindAdapter
+    params:
+      metrics:
+      - descriptorName: request_count
+        value: response.code | 100
+        labels:
+          source: source.name | target.name | source.name | target.name | "one"
+          target: source.name | target.name | source.name | "one"
+          service: target.name | api.name | target.name | "one"
+          method: api.method | target.name | api.method | target.name | "one"
+          response_code: response.code | response.code | response.code | response.code | response.code | response.code | response.code | response.code | 111
+`
+)
+
+func registerNoopAdapter(r adapter.Registrar) {
+	r.RegisterMetricsBuilder(&fakeNoopAdapter{adapter.NewDefaultBuilder("noopMetricKindAdapter", "Publishes metrics", &types.Empty{})})
+}
+
+type fakeNoopAdapter struct {
+	adapter.DefaultBuilder
```

Comment:
Use the [noop adapter](https://github.com/istio/mixer/tree/master/adapter/noop) for this instead.